src/averageAges.js

```diff
@@ -55,7 +73,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? getPersonsGender(
+      people.filter(child =>
+        people.find(mother => child.mother === mother.name)), 'm')
+    : people.filter(child =>
+      people.find(mother => child.mother === mother.name));
```

Comment:
you repeat yourself with this filter. Better just filter to find children, and store it in one variable. Then use/or not use your `getPersonsGender` on this var to find/not find sons